### ISSUE TITLE
Installation of certain packages fail on Windows

### ISSUE THREAD

**Issue by tkshnwesper** [luarocks/luarocks#730]
Here's the error that I get:
```
C:\Users\Saurabh\Code\Lua>luarocks install luafilesystem
Installing http://www.luarocks.org/repositories/rocks/luafilesystem-1.6.3-2.src.rock...
Using http://www.luarocks.org/repositories/rocks/luafilesystem-1.6.3-2.src.rock... switching to 'build' mode
mingw32-gcc -O2 -IC:/Users/Saurabh/lua/bin/../include -c src/lfs.c -o src/lfs.o
In file included from src/lfs.c:67:0:
src/lfs.h:21:0: warning: "fileno" redefined
 #define fileno(f) (_fileno(f))
 ^
In file included from src/lfs.c:38:0:
c:\mingw\include\stdio.h:854:0: note: this is the location of the previous definition
 #define fileno(__F) ((__F)->_file)
 ^
mingw32-gcc -shared -o lfs.dll src/lfs.o C:/Users/Saurabh/lua/bin/../lib -lmsvcr80
c:/mingw/bin/../lib/gcc/mingw32/5.3.0/../../../../mingw32/bin/ld.exe: cannot find C:\Users\Saurabh\lua\bin\..\lib: Permission denied
collect2.exe: error: ld returned 1 exit status

Error: Build error: Failed installing lfs.dll in C:\Users\Saurabh\lua\bin/../share/luarocks/rocks/lib/luarocks/rocks/luafilesystem
/1.6.3-2/lib
```
I installed another package ([lunajson](https://luarocks.org/modules/grafi/lunajson)) which installed correctly. Had posted the same [issue](https://github.com/keplerproject/luafilesystem/issues/95) in LFS, they asked me to post it here.

My Lua installation dir:
```
 Directory of c:\users\saurabh\lua

15-09-2017  19:00    <DIR>          .
15-09-2017  19:00    <DIR>          ..
08-02-2013  01:58    <DIR>          bin
08-02-2013  01:58    <DIR>          include
08-02-2013  01:58    <DIR>          lib
08-02-2013  01:58    <DIR>          share
15-09-2017  19:00    <DIR>          tmp
               0 File(s)              0 bytes
               7 Dir(s)  807,277,617,152 bytes free
```

Luarocks: 2.0.10
Lua: 5.1.5

**Comment by hishamhm** [comment]
Could you try it with the latest release of LuaRocks? (2.4.3)

**Comment by tkshnwesper** [comment]
I had downloaded luarocks for windows from [here](https://github.com/luarocks/luarocks/wiki/Installation-instructions-for-Windows) just a few days ago. How do I get the latest version for windows?

**Comment by daurnimator** [comment]
That page links to http://luarocks.github.io/luarocks/releases which has the 2.4.3 release available. http://luarocks.github.io/luarocks/releases/luarocks-2.4.3-win32.zip is the exact download link.

How did you get 2.0.10 before?

**Comment by tkshnwesper** [comment]
Apologies for my late reply. The reason my Luarocks version was old was because I had got it bundled with LuaDist. So when I typed in `luarocks` in my command line, it was calling the version from my LuaDist folder (since I had set its path).

tldr;
I installed the latest version and fixed all that after realizing my mistake and installed luafs successfully! Thanks :)
